src/components/KeyboardNavigation/Provider.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Provider } from "./Context";
+
+const ARROWUP = 38;
+const ARROWDOWN = 40;
+const SPACEBAR = 32;
+const TABKEY = 9;
+const ENTERKEY = 13;
+
+const WHITELIST = [ARROWUP, ARROWDOWN, SPACEBAR, TABKEY, ENTERKEY];
+
+export default class KeyBoardProvider extends React.Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired
+  };
+
+  static defaultProps = {};
+
+  onClick = ({ index }) => {
+    this.setState(() => ({ focused: index }));
+  };
+
+  onKeyDown = event => {
+    if (!WHITELIST.includes(event.keyCode)) return;
+
+    /* eslint-disable */
+
+    ![TABKEY, ENTERKEY, SPACEBAR].includes(event.keyCode) &&
+      event.preventDefault(); // eslint-disable-line
+
+    /* eslint-enable */
+
+    this.cycleSelect(event.keyCode);
+  };
+
+  cycleSelect = key => {
+    const { focused } = this.state;
+    const { children } = this.props;
+
+    if (key === ARROWUP && focused <= 0) {
+      this.setState(() => ({ focused: children.length - 1 }));
+    }
+
+    if (key === ARROWUP && focused > 0) {
+      this.setState(prevState => ({ focused: prevState.focused - 1 }));
+    }
+
+    if (key === ARROWDOWN && focused >= children.length - 1) {
+      this.setState(() => ({ focused: 0 }));
+    }
+
+    if (key === ARROWDOWN && focused < children.length - 1) {
+      this.setState(prevState => ({ focused: prevState.focused + 1 }));
+    }
+  };
+
+  /* eslint-disable */
+  state = {
+    focused: null,
+    focusSelected: this.onClick
+  };
+  /* eslint-enable */
+
+  render() {
+    return (
+      <Provider value={this.state}>
+        <div role="button" tabIndex="0" onKeyDown={this.onKeyDown}>
```

Comment:
let's remove this tabIndex I think it should be provided by a parent wrapping this component. also the `role` here is not needed.